src/spec.ts

```diff
@@ -101,8 +101,28 @@ export type Spec = UnitSpec | FacetSpec | LayerSpec;
 
 /* Custom type guards */
 
-export function isFacetSpec(spec: ExtendedSpec): spec is FacetSpec {
-  return spec['facet'] !== undefined;
+export function isFacetSpec(spec: ExtendedSpec | ExtendedFacetSpec): spec is FacetSpec {
+  if (spec['facet'] !== undefined) {
```

Comment:
I don't understand why you are modifying this method.   It seems irrelevant to the PR and will become incorrect in the future.